app/javascript/packs/homePageFeed.jsx

```diff
@@ -88,10 +89,15 @@ export const renderFeed = (timeFrame) => {
           sendFeaturedArticleAnalytics(featuredStory.id);
         }
 
+        const { currentUser } = getUserDataAndCsrfToken();
```

Comment:
@aitchiss Please check that I am using getUserDataAndCsrfToken correctly here 😅

Reply:
the only issue here is that `getUserDataAndCsrfToken` is async, so we need to wait for it to return.

I would suggest moving this line up to line 65 (so outside of the `<Feed>` component 'renderFeed' method), and awaiting it (we'll need to add `async` to the method signature too), so it would look like this:

```jsx
export const renderFeed = async (timeFrame) => {
  const feedContainer = document.getElementById('homepage-feed');

  const { currentUser } = await getUserDataAndCsrfToken();
  const currentUserId = currentUser && currentUser.id;

  render(
    <Feed
```